src/sentry/taskworker/worker.py

```diff
@@ -152,16 +169,12 @@ def handle_alarm(signum: int, frame: Any) -> None:
 
         set_current_task(activation)
 
-        # Set an alarm for the processing_deadline_duration
-        signal.alarm(activation.processing_deadline_duration)
-
-        execution_start_time = time.time()
         next_state = TASK_ACTIVATION_STATUS_FAILURE
+        execution_start_time = time.time()
```

Comment:
`time.time` should really never be used for timing as it is subjected to smear / etc.

you probably want `time.monotonic()` https://youtu.be/hb-12mgQrzM

Reply:
While I can switch a few usage of `time.time()`, I don't think all of them can be changed. We need to do some time differences against the message produce time which is a protobuf timestamp. These won't be precise measurements as clock times can vary from one system to another, but there aren't many other portable time representations.